Add codelayer-experimental cask for continuous deployment builds

K-Mistele opened the pull request:
## Summary

This PR adds a new Homebrew cask for experimental builds of CodeLayer that are automatically created from the main branch of the synclayer repository.

## Details

- **Cask name**: `codelayer-experimental`
- **App name**: CodeLayer-Experimental.app
- **Port**: 7779 (isolated from stable=7777, nightly=7778)
- **CLI commands**: `humanlayer-experimental`, `hld-experimental`
- **Database**: `~/.humanlayer/daemon-experimental.db`
- **Socket**: `~/.humanlayer/daemon-experimental.sock`

## Installation

Once merged, users will be able to install experimental builds with:

```bash
brew tap humanlayer/humanlayer
brew install --cask codelayer-experimental
```

## Side-by-side Installation

This experimental build can be installed alongside stable and nightly versions without conflicts, using completely isolated:
- Application bundles
- Configuration directories
- Database files
- Network ports
- CLI command names

## Automation

The cask will be automatically updated by GitHub Actions whenever a new experimental build is created from pushes to the main branch in the synclayer repository.

K-Mistele committed:
Add codelayer-experimental cask for continuous deployment builds

balanceiskey commented on Casks/codelayer-experimental.rb:
```diff
@@ -0,0 +1,28 @@
+
+  app "CodeLayer-Experimental.app"
+
+  binary "#{appdir}/CodeLayer-Experimental.app/Contents/Resources/bin/humanlayer", target: "humanlayer-experimental"
```
Do we want to add a `codelayer-experimental` flag here (we may need to update this as well, which has already been merged: https://github.com/humanlayer/humanlayer/pull/795)

K-Mistele replied:
I am not sure what that means or what the implication of that would be :/

K-Mistele replied:
claude is spinning on that for me - am I correct in understanding this would need to be a change in the new repo that this cask is coming from?

balanceiskey replied:
It's minor: When we build `codelayer` and `codelayer-nightly` today we also provide symlinks to the `hlyr` CLI via these `binary` lines with a `target`, so you automatically get `codelayer` and `codelayer-nightly` at the CLI level (mostly to open the app from the CLI or launch a session). The merged PR above also adds functionality to the original `hlyr` CLI that checks if `codelayer` or `codelayer-nightly` are part of the invoked name and reach for either the standard app build or the nightly build depending on which was called.

K-Mistele commented:
Thanks for the feedback! I've addressed it by adding  support to the synclayer hlyr source:

**What I did:**
1. Synced the latest hlyr source from humanlayer to synclayer (to get the smart invocation detection from PR #795)
2. Added `codelayer-experimental` to the bin entries in package.json
3. Updated the invocation detection code to handle the experimental variant:
   - Uses `~/.humanlayer/daemon-experimental.sock`
   - Opens `CodeLayer-Experimental.app` when called without arguments
   - Works as CLI when called with arguments

**Commit:** https://github.com/humanlayer/synclayer/commit/663e7db

This ensures the experimental build has the same smart CLI behavior as stable and nightly builds. The cask is now ready to work with the smart invocation detection!
